Lib/test/_test_multiprocessing.py

```diff
@@ -361,6 +361,27 @@ def test_process(self):
         self.assertNotIn(p, self.active_children())
         close_queue(q)
 
+    if threading._HAVE_THREAD_NATIVE_ID:
```

Comment:
I think it's more conventional here to use `@unittest.skipif(not threading._HAVE_THREAD_NATIVE_ID, "don't have native_id")` instead of this branching. Then both of these methods can be unindented.

`_test_process_thread_attributes` can also be a `@staticmethod` (`cls` isn't used). And personally, I would recommend renaming it to something less test-like and more unique... but that's just me!

Reply:
On the first, I was following the convention in the `threading` module but this seems more reasonable. 

While `_test_process_thread_attributes` could be a `staticmethod`, I would prefer to keep consistency within this class – the other target functions all use `@classmethod` regardless if they use the `cls` variable or not. 
I should probably rename the function though, good point.